packages/browser/src/index.ts

```diff
@@ -88,6 +93,75 @@ export default class LogtoClient {
     sessionStorage.setItem(this.logtoStorageKey, jsonItem);
   }
 
+  public async getAccessToken(resource?: string): Promise<Optional<string>> {
+    if (!this.idToken) {
+      throw new LogtoClientError('not_authenticated');
+    }
+
+    const accessTokenKey = buildAccessTokenKey(resource);
+    const accessToken = this.accessTokenMap.get(accessTokenKey);
+
+    if (!accessToken) {
+      throw new LogtoClientError('get_access_token_failed');
+    }
+
+    if (accessToken.expiresAt < Date.now()) {
+      this.accessTokenMap.delete(accessTokenKey);
+
+      const { clientId, requester } = this.logtoConfig;
+      const { tokenEndpoint } = await this.getOidcConfig();
+
+      if (!this.refreshToken) {
+        throw new LogtoClientError('not_authenticated');
+      }
+
+      try {
+        const { accessToken, refreshToken, idToken, scope, expiresIn } =
+          await fetchTokenByRefreshToken(
+            { clientId, tokenEndpoint, refreshToken: this.refreshToken, resource },
+            requester
+          );
+        this.accessTokenMap.set(accessTokenKey, {
+          token: accessToken,
+          scope,
+          expiresAt: Math.round(Date.now() / 1000) + expiresIn,
+        });
+
+        localStorage.setItem(`${this.logtoStorageKey}:refreshToken`, refreshToken);
+
+        if (idToken) {
+          localStorage.setItem(`${this.logtoStorageKey}:idToken`, idToken);
```

Comment:
@xiaoyijun do we need to validate the `idToken` here?

Reply:
Yes we do need to verify idToken here, will update shortly